File: js/src/ArrayProblem/SimpleBinarySearch.js



export function SimpleBinarySearch(list ,target){

		let left = 0;
		let right = list.length - 1;


		while(left <= right){

			let mid = Math.floor((left + right) / 2);

			if (list[mid] == target) return mid;

			if (list[mid] < target) {

				left = mid  + 1;

			}else{
				right = mid - 1;
			}

		}

		return -1;
}

File: python/array_problem/simple_binary_search.py




def find_by_simple_binary_Search(listx, target):

	left = 0;
	right = len(listx) - 1

	while left <= right:
		
		mid = left + (right - left)

		if listx[mid] == target:
			return mid

		elif listx[mid] < target:

			left = mid + 1;
		else:
			right = mid -1;

	return -1;